multicluster/cmd/multicluster-controller/memberclusterannounce_webhook.go

```diff
@@ -67,20 +74,20 @@ func (v *memberClusterAnnounceValidator) Handle(ctx context.Context, req admissi
 	}
 
 	clusterSet := clusterSetList.Items[0]
-	if clusterSet.Name == memberClusterAnnounce.ClusterSetID {
-		for _, member := range clusterSet.Spec.Members {
-			if member.ClusterID == memberClusterAnnounce.ClusterID {
-				// validate the ServiceAccount used is correct
-				if member.ServiceAccount == saName {
-					return admission.Allowed("")
-				} else {
-					return admission.Denied("Member does not have permissions")
-				}
-			}
-		}
+	if len(clusterSet.Spec.Leaders) != 1 {
+		return admission.Errored(http.StatusPreconditionFailed,
+			fmt.Errorf("invalid ClusterSet config in the leader cluster, there must be one leader cluster in the ClusterSet"))
 	}
 
-	return admission.Denied("Unknown member")
+	if clusterSet.Name == memberClusterAnnounce.ClusterSetID &&
```

Comment:
I feel the following is easier to understand:

```
	if clusterSet.Name != memberClusterAnnounce.ClusterSetID {
		return admission.Denied("Unknown ClusterSet ID")
	}

	if clusterSet.Spec.Leaders[0].ClusterID != memberClusterAnnounce.LeaderClusterID {
		return admission.Denied("Leader cluster ID in the MemberClusterAnnounce does not match that in the ClusterSet")
	}
	return admission.Allowed("")
```

Reply:
Fixed.